dune/grid/io/file/printgrid.hh

```diff
@@ -0,0 +1,191 @@
+                int size = 2000, bool execute_plot = true, bool png = true, bool local_corner_indices = true,
+                bool local_intersection_indices = true, bool outer_normals = true)
+{
+
+  // Create output file
+  output_file = output_file + "_" + std::to_string(helper.rank());
+  std::ofstream plotfile (output_file, std::ios::out | std::ios::trunc);
+  if (!plotfile.is_open()) {
+    std::cerr << "Could not create plot file" << std::endl;
+    return;
+  }
+
+  // Basic plot settings
+  plotfile << "set size ratio -1" << std::endl;
+  if (png) {
+    plotfile << "set terminal png size " << size << "," << size << std::endl;
+    plotfile << "set output '" << output_file << ".png'" << std::endl;
+  } else {
+    plotfile << "set terminal svg size " << size << "," << size << " enhanced background rgb 'white'" << std::endl;
+    plotfile << "set output '" << output_file << ".svg'" << std::endl;
+  }
+
+  // Get GridView
+  typedef typename GridType::LeafGridView GV;
+  const GV& gv=grid.leafGridView();
+
+  // Create mappers used to retrieve indices
+  typedef typename Dune::LeafMultipleCodimMultipleGeomTypeMapper<GridType, ElementDataLayout      > ElementMapper;
+  typedef typename Dune::LeafMultipleCodimMultipleGeomTypeMapper<GridType, NodeDataLayout         > NodeMapper;
+  ElementMapper elementmapper(grid);
+  NodeMapper nodemapper(grid);
+
+  // Create iterators
+  typedef typename GV::template Codim<0 >::Iterator LeafIterator;
+  typedef typename GV::IntersectionIterator IntersectionIterator;
+
+  LeafIterator it = gv.template begin<0>();
+
+  // Will contain min/max coordinates. Needed for scaling of the plot
+  Dune::FieldVector<typename GridType::ctype,2> max_coord (it->geometry().center()), min_coord (max_coord);
+
+  // Iterate over elements
+  for (; it != gv.template end<0>(); ++it) {
+
+    auto geo = it->geometry();
+
+    // Plot element index
+    int element_id = elementmapper.index(*it);
+    plotfile << "set label at " << geo.center()[0] << "," << geo.center()[1] << " '"
+          << element_id << "' center" << std::endl;
+
+    for (int i = 0; i < geo.corners(); ++i) {
+      // Plot corner indices
+      const int globalNodeNumber1 = nodemapper.subIndex(*it, i, 2);
+      auto labelpos = centrify (geo, geo.corner(i), 0.7);
+      plotfile << "set label at " << labelpos[0] << "," << labelpos[1] << " '" << globalNodeNumber1;
+      if (local_corner_indices)
+        plotfile << "(" << i << ")";
+      plotfile << "' center" << std::endl;
+
+      // Adapt min / max coordinates
+      for (int dim = 0; dim < 2; ++dim) {
+        if (geo.corner(i)[dim] < min_coord[dim])
+          min_coord[dim] = geo.corner(i)[dim];
+        else if (geo.corner(i)[dim] > max_coord[dim])
+          max_coord[dim] = geo.corner(i)[dim];
+      }
+    }
+
+    // Iterate over intersections
+    for (IntersectionIterator is = gv.ibegin(*it); is != gv.iend(*it); ++is) {
+
```

Comment:
Same thing as for the entity: Store a const ref to the intersection at the beginning of the loop